Pub/Sub: raise exceptions when invalid id

gctucker opened the pull request:
Raise a `ValueError` exception when the subscription id is not valid.

JenySadadia reviewed (changes requested):
We also need to catch ValueError in `listen` endpoint.

```
@app.get('/listen/{sub_id}')
async def listen(sub_id: int, user: User = Depends(get_user)):
    try:
        msg = await pubsub.listen(sub_id)
        if msg is None:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND,
                detail=f"Not subscribed to channel with id: {sub_id}"
            )
        return msg
    except ValueError as error:
        raise HTTPException(
                status_code=status.HTTP_204_NO_CONTENT,
                detail=str(error)
        )
```

gctucker commented:
> We also need to catch ValueError in `listen` endpoint.
> 
> ```
> @app.get('/listen/{sub_id}')
> async def listen(sub_id: int, user: User = Depends(get_user)):
>     try:
>         msg = await pubsub.listen(sub_id)
>         if msg is None:
>             raise HTTPException(
>                 status_code=status.HTTP_404_NOT_FOUND,
>                 detail=f"Not subscribed to channel with id: {sub_id}"
>             )
>         return msg
>     except ValueError as error:
>         raise HTTPException(
>                 status_code=status.HTTP_204_NO_CONTENT,
>                 detail=str(error)
>         )
> ```

OK I've added that and updated unit tests accordingly.

gctucker commented:
@JenySadadia Could you please take another look?

JenySadadia commented on test/test_listen_handler.py:
```diff
@@ -108,7 +110,8 @@ def test_listen_endpoint_without_token(mock_get_current_user,
                 active=True)
     mock_get_current_user.return_value = user
-    mock_listen.return_value = 'Listening for events on channel 1'
+    mock_listen.return_value = None
```
We can omit `mock_listen` from this.

```
def test_listen_endpoint_without_token(mock_get_current_user,
                                       mock_init_sub_id):
    """
    Test Case : Test KernelCI API GET /listen endpoint for the
    negative path
    Expected Result :
        HTTP Response Code 401 Unauthorized
        The request requires user authentication by token in header
    """
    user = User(username='bob',
                hashed_password='$2b$12$CpJZx5ooxM11bCFXT76/z.o6HWs2sPJy4iP8.'
                                'xCZGmM8jWXUXJZ4K',
                active=True)
    mock_get_current_user.return_value = user

    with TestClient(app) as client:
        response = client.get(
            "/listen/1",
            headers={
                "Accept": "application/json"
            },
        )
        assert response.status_code == 401
```

JenySadadia commented on test/test_listen_handler.py:
```diff
@@ -68,22 +68,24 @@ def test_listen_endpoint(mock_get_current_user,
 
 
 def test_listen_endpoint_not_found(mock_get_current_user,
```
We can add another assert statement to endpoint `test_listen_endpoint_not_found`:
```
assert 'detail' in response.json()
```
